util: update math.go

Anudhyan opened the pull request:
<!--

Thank you for contributing to TiDB!

PR Title Format:
1. pkg [, pkg2, pkg3]: what's changed
2. *: what's changed

-->

### What problem does this PR solve?
<!--

Please create an issue first to describe the problem.

There MUST be one line starting with "Issue Number:  " and
linking the relevant issues via the "close" or "ref".

For more info, check https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/contribute-code.html#referring-to-an-issue.

-->

Issue Number: close #56594

Problem Summary:
- Swapping out `mathutil` functions in favor of Go's built-in ones, like trading in a custom `Max` function for Go's `math.Max` 🛠️.

### What changed and how does it work?

- The code now uses Go's built-in functions (bye-bye `mathutil`), making things simpler and cleaner.

### Check List

Tests <!-- At least one of them must be included. -->

- [ ] Unit test
- [ ] Integration test
- [x] Manual test (add detailed scripts or steps below)
  > - Gave it a good ol' manual test to make sure everything runs smooth like butter with the built-in functions.

Side effects

- [ ] Performance regression: Consumes more CPU
- [ ] Performance regression: Consumes more Memory
- [ ] Breaking backward compatibility

Documentation

- [ ] Affects user behaviors
- [x] Contains syntax changes
- [x] Contains variable changes
- [ ] Contains experimental features
- [ ] Changes MySQL compatibility

### Release note

<!-- compatibility change, improvement, bugfix, and new feature need a release note -->

Please refer to [Release Notes Language Style Guide](https://pingcap.github.io/tidb-dev-guide/contribute-to-tidb/release-notes-style-guide.html) to write a quality release note.

```release-note
None

Anudhyan committed:
Update math.go

Removed the custom Max and Min functions.
Updated the Clamp function to use built-in max and min.
Removed the import for golang.org/x/exp/constraints as it's no longer needed.

lance6716 commented:
/ok-to-test

Anudhyan commented:
> /ok-to-test

@lance6716 could you please suggest me what to do?

lance6716 commented:
> > /ok-to-test
> 
> @lance6716 could you please suggest me what to do?

Please make this PR compile. You can run `make server` at the project root folder.

Anudhyan committed:
Update math.go

Anudhyan commented:
/retest

Anudhyan committed:
Update math.go

hawkingrei commented:
@Anudhyan Your modifications have compilation issues.

Anudhyan committed:
Update math.go

Anudhyan committed:
Update math.go

Anudhyan committed:
math.go

I had added the built in function and removed the manual function. The code is tested

Anudhyan committed:
Update math.go
